synapse/rest/client/room.py

```diff
@@ -204,6 +204,11 @@ async def on_PUT(
         if state_key is not None:
             event_dict["state_key"] = state_key
 
+        # Twisted will have processed the args by now.
+        assert request.args is not None
+        if b"ts" in request.args and requester.app_service:
+            event_dict["origin_server_ts"] = parse_integer(request, "ts", 0)
```

Comment:
I think `None` would make more sense as a default value for `parse_integer`, as that implies the parameter could not be found. Whereas `0` is (technically) a valid value for `origin_server_ts`.

Note with that, you'd no longer need the `b"ts" in request.args` conditional check:

```python
        if requester.app_service:
            event_dict["origin_server_ts"] = parse_integer(request, "ts")
```

Reply:
Done :)